Orange/widgets/data/owselectrows.py

```diff
@@ -484,22 +523,49 @@ def add_datetime(contents):
             box = gui.hBox(self, addToLayout=False)
             box.var_type = vtype
             self.cond_list.setCellWidget(oper_combo.row, 2, box)
-            if vtype in (2, 4):  # continuous, time:
-                validator = add_datetime if isinstance(var, TimeVariable) else add_numeric
-                box.controls = [validator(lc[0])]
+            if vtype == 2:  # continuous:
+                box.controls = [add_numeric(lc[0])]
                 if oper > 5:
                     gui.widgetLabel(box, " and ")
-                    box.controls.append(validator(lc[1]))
+                    box.controls.append(add_numeric(lc[1]))
             elif vtype == 3:  # string:
                 box.controls = [add_textual(lc[0])]
                 if oper in [6, 7]:
                     gui.widgetLabel(box, " and ")
                     box.controls.append(add_textual(lc[1]))
+            elif vtype == 4:  # time:
+                datetime_format = (var.have_date, var.have_time)
+                w = DateTimeWidget(self, var_idx, datetime_format)
+                w.set_datetime(lc[0])
+                box.controls = [w]
+                box.layout().addWidget(w)
+                self._box = box
+                if oper > 5:
+                    gui.widgetLabel(box, " and ")
+                    w_ = DateTimeWidget(self, var_idx, datetime_format)
+                    w_.set_datetime(lc[1])
+                    box.layout().addWidget(w_)
+                    box.controls.append(w_)
+                    self._box = box
+                    self._invalidate_dates()
             else:
                 box.controls = []
         if not adding_all:
             self.conditions_changed()
 
+    def _invalidate_dates(self):
+        w = self._box.controls[0]
+        if len(self._box.controls) > 1:
+            w_ = self._box.controls[1]
+            if w.dateTime() > w_.dateTime():
+                w_.setDateTime(w.dateTime())
+                w_.dateTimeChanged.connect(self.conditions_changed)
```

Comment:
Huh, this reconnects the signal at every change?! You wanted to connect the method or to just call it?